jenkins/spark-premerge-build.sh

```diff
@@ -43,17 +43,20 @@ mvn_verify() {
     env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=311cdh clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
     env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=312 clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
     env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=313 clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
-    env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=314 clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
+    # skip spark snapshot shims build in 22.04. (TODO) revert this commit in 22.06 after auto-merged
```

Comment:
Let us formalize when the snapshots build are disabled so we don't have to do it over and over manually by commenting and uncommenting. This could be an environment variable or a dedicated configuration file in the repo. 

Maybe we can reuse .github/workflows/auto-merge.yml ? Whether the snapshot builds are needed is directly related to auto-merge as this comment indicates. Could we  make the current branch name's equality to the `BASE` in .github/workflows/auto-merge.yml

```bash
BASE_BRANCH=$(grep 'BASE: ' .github/workflows/auto-merge.yml | grep -o branch-.*)
CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)

if [[ "$CURRENT_BRANCH" == "$BASE_BRANCH" ]]; then
  # build snapshots
  ...
fi
```

Reply:
If so, we could simply based on https://github.com/NVIDIA/spark-rapids/blob/branch-22.06/jenkins/version-def.sh#L31 instead of a workflow file. let me refactor this